Create LICENSE file

d3zd3z opened the pull request:
The code is under quite a few licenses, try to collect them into a
single file.

jenswikl commented:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

pascal-brand38 commented on LICENSE:
```diff
@@ -0,0 +1,264 @@
+
+----------------------------------------------------------------------
+
+bios/arm32.h:
```
Should we detail here each file and each owner? This will become difficult to support and keep up-to -date. I would prefer to indicate there is GPLv2 / BSD2 / BSD3 / NEWLIB / dlmalloc public domain /  Smartall public domain license terms.

@jbech-linaro what do you think?

jockebech replied:
I think it'd be nice to have a single LICENSE file that tells about all licenses in use and then in each and every file and the actual license. I think we've done that in most of our own files so far. From what I recall with a discussion with SY, it's probably only newlib related files that haven't any licenses within the file(s).

d3zd3z replied:
There is only one file mentioned in the LICENSE file, the bios/arm32.h. There is also a single newlib file covered under that license.

d3zd3z commented:
Let's try this way. I've just grabbed the COPYING.NEWLIB file directly, and the COPYING file contains the license texts extracted, but without any mention of the files involved.

jockebech commented:
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

d3zd3z committed:
Create license files

The code is under quite a few licenses, collect them into a single
file.  Also, since a single file is covered under the default newlib,
license, include the copying file from that project as well.

Signed-off-by: David Brown <david.brown@linaro.org>
Reviewed-by: Joakim Bech <joakim.bech@linaro.org>

d3zd3z commented:
I've updated this, and added Joakim's sob.

ghost commented:
Closing as manually merged
